File: October 2022/Day 29/C++ Code/2136. Earliest Possible Day of Full Bloom.cpp
class Solution {
public:
    // T.C :- O(n.log(n))
    int earliestFullBloom(vector<int>& plantTime, vector<int>& growTime) {
        vector<pair<int,int>> a;
        for(int i=0;i<plantTime.size();i++) {
            pair<int,int> p(plantTime[i],growTime[i]);
            a.push_back(p);
        }
        
        sort(a.begin(),a.end(),comp);
        
        int ans=0;
        int worked_day=0;
        for(int i=0;i<a.size();i++) {
            worked_day+=a[i].first;
            ans = max(ans,worked_day+a[i].second);
        }
        
        return ans;
    }
    
    static bool comp(const pair<int,int>& p1,const pair<int,int>& p2) {
        return p1.second>p2.second;
    }
};

File: October 2022/Day 29/JAVA Code/2136. Earliest Possible Day of Full Bloom.java
class Solution {
    
    // Time Complexity :- O(n.log(n));
    
    public int earliestFullBloom(int[] plantTime, int[] growTime) {
        int n = plantTime.length;
        int[][] a = new int[n][2];
        for(int i=0;i<n;i++) {
            a[i][0]=plantTime[i];
            a[i][1]=growTime[i];
        }
        
        // sort in descending order according to growTime i.e index 1st of a[];
        Arrays.sort(a,(b,c) -> (c[1]-b[1]));
        
        int ans=0;
        int worked_day=0;
        for(int i=0;i<n;i++) {
            worked_day+=a[i][0];
            ans = Math.max(ans,worked_day+a[i][1]);
        }
        
        return ans;
    }
}
